mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java

```diff
@@ -170,27 +169,53 @@ public void service(Mail mail) throws MessagingException {
     }
 
     private void setAttribute(Mail mail, Map<String, Calendar> calendars, Map<String, byte[]> rawCalendars) throws MessagingException {
-        Optional<String> sender = retrieveSender(mail);
+        Optional<MailAddress> sender = retrieveSender(mail);
         if (!sender.isPresent()) {
             LOGGER.info("Skipping {} because no sender and no from", mail.getName());
             return;
         }
+
+        MailAddress transportSender = sender.get();
+        MailAddress replyTo = fetchReplyTo(mail).orElse(transportSender);
+
         Map<String, byte[]> jsonsInByteForm = calendars.entrySet()
             .stream()
-            .flatMap(calendar -> toJson(calendar, rawCalendars, mail, sender.get()))
+            .flatMap(calendar -> toJson(calendar, rawCalendars, mail, transportSender, replyTo))
             .collect(Guavate.toImmutableMap(Pair::getKey, Pair::getValue));
         mail.setAttribute(new Attribute(destinationAttributeName, AttributeValue.ofAny(jsonsInByteForm)));
     }
 
-    private Stream<Pair<String, byte[]>> toJson(Map.Entry<String, Calendar> entry, Map<String, byte[]> rawCalendars, Mail mail, String sender) {
+    private Optional<MailAddress> fetchReplyTo(Mail mail) throws MessagingException {
+        return Optional.ofNullable(mail.getMessage())
+            .flatMap(Throwing.<MimeMessage, Optional<String[]>>function(mimeMessage ->
+                    Optional.ofNullable(mimeMessage.getHeader(REPLY_TO_HEADER_NAME))
+                ).sneakyThrow())
+            .filter(headers -> headers.length > 0)
+            .map(headers -> headers[0])
+            .flatMap(this::retrieveReplyTo);
+    }
+
+    private Optional<? extends MailAddress> retrieveReplyTo(String headerValue) {
+        try {
+            return Optional.of(new MailAddress(new InternetAddress(headerValue)));
```

Comment:
I'm not fan of relying on javax.mail in yet another place for address header parsing.

How about:

```
LenientAddressParser.DEFAULT
                .parseAddressList(rawHeaderValue)
                .stream()
                .flatMap(this::convertAddressToMailboxStream)
                .map(mailbox -> new MailAdress(mailbox.getAddress()))
                .findFirst();
```

&&

```
        private Stream<Mailbox> convertAddressToMailboxStream(Address address) {
            if (address instanceof Mailbox) {
                return Stream.of((Mailbox) address);
            } else if (address instanceof Group) {
                return ((Group) address).getMailboxes().stream();
            }
            return Stream.empty();
        }
```

Reply:
(code samples from  org.apache.james.mailbox.elasticsearch.json.HeaderCollection)

Bonus: group handling (syntax: `group: john.doe@acme.org, Mary Smith <mary@example.net>` taken from LenientAddressBuilderTest